gpu.h

```diff
@@ -591,6 +594,53 @@ inline Tensor createTensor(Context &ctx, const Shape &shape, NumType dtype,
   return tensor;
 }
 
+/**
+ * @brief Overload of the tensor factory function to instantiate a tensor on
+ * the GPU with a given shape, data type. This overload also takes initial
+ * half* data to populate the tensor with.
+ *
+ * The data is assumed to be of size equal to the product of the dimensions in
+ * the shape, and is copied to the GPU buffer.
+ *
+ * @param[in] ctx Context instance to manage the tensor
+ * @param[in] shape Shape of the tensor
+ * @param[in] data Initial data to populate the tensor with
+ * @return Tensor instance representing the created tensor
+ *
+ * @code
+ * Tensor tensor = createTensor(ctx, {256, 256}, data);
+ * @endcode
+ */
+template<class precision>
+Tensor createTensor(Context &ctx, const Shape &shape, precision *data) {
```

Comment:
as with other comments, let's keep `createTensor` at value level unless there's something we really need to add the templated variants.